theano/gof/opt.py

```diff
@@ -3039,3 +3039,31 @@ def check_stack_trace(f_or_fgraph, ops_to_check='last', bug_print='raise'):
                 return False
 
     return True
+
+
+class CheckStrackTraceFeature(object):
+    def on_import(self, fgraph, node, reason):
+        if theano.config.check_stack_trace == 'check_all':
+            if not check_stack_trace(fgraph, 'all'):
+                raise NotImplementedError(
+                    "Empty stack trace! The optimization that whose"
+                    " stacktrace is empty is" + str(reason))
```

Comment:
I would rather say `"Empty stack trace! The optimization that inserted that variable is " + str(reason)`.